yt/frontends/enzo/data_structures.py

```diff
@@ -980,10 +981,10 @@ def _set_code_unit_attributes(self):
         setdefaultattr(self, "magnetic_unit", self.quan(magnetic_unit, "gauss"))
 
     @classmethod
-    def _is_valid(cls, filename, *args, **kwargs):
-        if str(filename).endswith(".hierarchy"):
-            return True
-        return os.path.exists(f"{filename}.hierarchy")
+    def _is_valid(cls, filename: str, *args, **kwargs) -> bool:
+        return filename.endswith(".hierarchy") or os.path.exists(
+            f"{filename}.hierarchy"
+        )
```

Comment:
OK, odd quirk here! I just tried was loading `IsolatedGalaxy` in a fresh environment with a minimal yt install (so no h5py), and you get:

```
>>> ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
<stdin>:1: UserWarning: This dataset appears to be of type EnzoDataset, but the following requirements are currently missing: h5py, libconf
Please verify your installation.
yt : [INFO     ] 2023-07-20 11:34:36,033 Parameters: current_time              = 0.0060000200028298
yt : [INFO     ] 2023-07-20 11:34:36,033 Parameters: domain_dimensions         = [32 32 32]
yt : [INFO     ] 2023-07-20 11:34:36,033 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2023-07-20 11:34:36,034 Parameters: domain_right_edge         = [1. 1. 1.]
yt : [INFO     ] 2023-07-20 11:34:36,034 Parameters: cosmological_simulation   = 0
```
Based on the `_is_valid` here, the fact that the load appears to work (you actually get populated parameters!) would not be new to this PR , so it's actually quite helpful to have that `UserWarning` show up. But would it make sense to add an error here? 

Initially I was gonna suggest adding

```python
        if cls._missing_load_requirements():
            return False
```
like in the other updated `_is_valid` methods. But then that would print out the full list of h5py-dependent frontends that are possible. And here we actually know it's enzo.

Reply:
I think the paradigm has always been that `_is_valid` methods are not allowed to throw errors as it would give each frontend the power to break all others. Maybe I'm not understanding your suggestion ?

Reply:
Not suggesting to throw an error within the method. 

Just saying that you could refactor this is_valid to return False if the dependencies are missing, regardless of the filename convention check that it does now. That would result in a unidentified datatype error right away rather than a partially functional `ds`.

Reply:
I'm just on the fence on whether that it's better have a non functional ds but know that it's Enzo or an error right away and not know it's Enzo.

Reply:
oh, I see. The behaviour you're pointing to is not intended ! I must have been testing this with `yt.load_sample("IsolatedGalaxy")` instead, which behaves differently. Your suggestion makes both loaders behave similarly, which is great, but we also loose the warning... I'll switch to draft while I try to think of a better solution.

Reply:
ah, I see now that I replied in a hurry: as you already pointed, the behaviour of the Enzo frontend is weird on the main branch, and my patch only adds a warning to it (which is the point).
I'd prefer to keep this PR's scope limited to adding warnings and details to existing errors, rather than actively changing frontends' behaviour, just to keep things as tidy as possible. But I would merge your suggestion as a standalone patch ! WDYT ?

Reply:
Ya, that sounds good to me!

Reply:
Mostly wanted to bring up this behavior --- even without a follow-up patch, having the warning now is better than the behavior on main!